rfp-proposals/ars-go-filecoin-storage-helper.md

```diff
@@ -0,0 +1,99 @@
+# RFP Proposal: `ars-go-filecoin-storage-helper`
+
+**Name of Project:** go-filecoin-storage-helper
+
+**Link to RFP:** [rfp-storage-helper-libraries](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md)
+
+**RFP Category:** `devtools-libraries`
+
+**Proposer:** [@waynewyang](https://github.com/waynewyang)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:**  "Yes"
+
+# Project Description
+
+The project is based on [rfp-storage-helper-libraries](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md), We think we're  very clear about the project want to do, our implementation will meet the need of [RFP](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-storage-helper-libraries.md),
+
+We think the keypoint is that client must pay for the metadata, the metadata is used for data retrieval,so the design of the metadata is very important,now we already have the idea.
+
+Besides,we have another idea we'll provide the function in storage helper that metadata encryption.
+
+## Deliverables
+
+- Meet the need of the original RFP,
+  - An open-sourced, dual-licensed (under MIT and APACHE2) library that performs a useful workflow and can be used modularly by other software applications in the Filecoin ecosystem
+  - Can be in any language, but we have a preference for Go
+  - Documentation of how to use and install this library
+  - Sample application (can be extremely simple) that uses both this helper library and go-filecoin to perform the intended workflow
+  - Well-documented, human-readable codebase.
+- Besides, we want to point out as belows
+  - Our implementation  will meet any size directory and file upload through the storage helper
+  - A large file or directory needs to be split into multiple deals, we'll implement these functions as belows,
+    - Deals monitors
+    - Callback of deals status and reminder that deadline of the deals
+    - Automatically maintain orders after the deadline of the deals reached
+  - Metadata encryption
+
+## Development Roadmap
+
+
+| Milestone No. | Milestone Description | Funding | Estimated Timeframe |
+| --- | --- | --- | --- |
+| 1 | Storage helper library | $3500 | 2 weeks |
+| 2 | All deliverables (codebase, sample application, documentation, testing) | $1500 | 1 week |
+
+
+## Total Budget Requested
+
+**Total Funding Amount:** $5000
+
+
+## Maintenance and Upgrade Plans
```

Comment:
Can you clarify whether you are planning to maintain this project on your own (i.e. without additional funding)?

Reply:
@pooja  yes,we'll maintenance work without additional funding.

Reply:
@pooja  yes,we'll maintenance work without additional funding.